### ISSUE TITLE
Exporting torch model with DFT results in transform being taken along the wrong dimension

### ISSUE THREAD

**Issue by jamied157** [microsoft/onnxscript#2704]
I'm using `onnxscript==0.5.4` and `torch==2.9.0` but expect this to still be a problem with the latest onnxscript as the code here: https://github.com/microsoft/onnxscript/blob/c1bfdfc5baf2dabf09191ad2ab7603ea21cfe614/onnxscript/function_libs/torch_lib/ops/fft.py hasn't  been updated for a while.

Exporting a model in pytorch like 
```
import torch.nn as nn
import torch

in_data = torch.randn(2, 128, 512)

class DFTModel(nn.Module):
    def __init__(self):
        super().__init__()
        
    def forward(self, a):
        return torch.fft.rfft(a)


torch.onnx.export(DFTModel(), (in_data,), "dft.onnx", dynamo=True)
```

Results in a model that looks like:

<img width="220" height="369" alt="Image" src="https://github.com/user-attachments/assets/29b6f6a6-40ff-45ba-94c6-900bcb70da24" />

Whilst the output shapes are correct here, the attributes of the model are not. We unsqueeze the final axis and then set the dft_length to 2. This results in the DFT being calculated on the wrong dimension (the extra trailing dimension with padding). I think the correct graph should look more like:

<img width="268" height="495" alt="Image" src="https://github.com/user-attachments/assets/673bc46c-e16c-47b3-90e7-9b9f6e6a45a2" />

To verify the model is wrong you can try and run it, I get this result

<img width="677" height="166" alt="Image" src="https://github.com/user-attachments/assets/57b59e39-36ab-418b-954f-d3bcefff6ea5" />

Whereas I get this with the correct model

<img width="688" height="129" alt="Image" src="https://github.com/user-attachments/assets/1cc5964c-30b7-4a20-98da-57e3d428d230" />

Happy to go in and patch this with some pointers!

**Comment by justinchuby** [comment]
When I look at the implementation we don’t supply dft_length in fft_r2c. Not sure where it is coming from 🤔

**Comment by jamied157** [comment]
Oh it's this: https://github.com/microsoft/onnxscript/commit/69025f7f7b746eb62292135be2eb3e00f1b36890 🤦
